as/cNFT/assembly/mint.ts

```diff
@@ -5,12 +5,21 @@ import { persistent_tokens } from './models/persistent_tokens'
 import { persistent_tokens_metadata, TokenMetadata } from './models/persistent_tokens_metadata'
 import { persistent_tokens_royalty, TokenRoyalty } from './models/persistent_tokens_royalty'
 import { NftEventLogData, NftMintLog } from './models/log'
+import { nft_metadata } from './metadata'
 
 @nearBindgen
 export function mint(tokenMetadata: TokenMetadata, token_royalty: TokenRoyalty): Token {
     /**@todo Assert deposit attached based on custom amount from init */
     //     assert_deposit_attached(u128.from(0))
 
+
+    const owner_supply = parseInt(persistent_tokens.supply_for_owner(context.sender))
```

Comment:
It's preferable to move this new code into a function with a name like:
```
assert_one_mint_per_address
```

Reply:
actually sorry, let's try to make this as generic as possible and call it `assert_mint_per_address` and pass the max mints per address to the contract `init` function.